File: smart-home-client/src/models/RealEstate.ts
import { Address } from "./Address";
import { LoginUser} from "./User";
import { RealEstateStatus } from "./enums/RealEstateStatus";



export interface RealEstate {
    id: number;
    name: string;
    address: Address;
    userId: LoginUser | null;
    quadrature: number;
    floors: number;
    picture: string;
    status: RealEstateStatus;
}

export interface NRealEstate {
    name: string;
    address: Address;
    userId: LoginUser | null;
    quadrature: number;
    floors: number;
    picture: string;
}


export interface RealEstateStatusChange {
    id: number;
    status: RealEstateStatus;
    reason: string

}

File: smart-home-server/src/main/java/com/example/smarthome/model/RealEstate.java
package com.example.smarthome.model;

import com.example.smarthome.model.enums.RealEstateStatus;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Entity
@Table(name = "real_estate")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class RealEstate implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Integer id;
    @Column(unique = true)
    private String name;
    @ManyToOne(cascade = CascadeType.MERGE, fetch = FetchType.EAGER)
    @JoinColumn(name = "address_id")
    private Address address;
    @ManyToOne(cascade = CascadeType.MERGE, fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id")
    private User userId;
    @Column
    private int quadrature;
    @Column
    private int floors;
    @Column
    private String picture;
    @Column
    private RealEstateStatus status;
}
